src/priority_fee.rs

```diff
@@ -193,58 +196,120 @@ impl GrpcConsumer for PriorityFeeTracker {
-                    "high_priority_fee",
-                    global_fees.high as u64,
-                    "account" => "none"
-                );
-                statsd_gauge!(
-                    "very_high_priority_fee",
-                    global_fees.very_high as u64,
-                    "account" => "none"
-                );
-                statsd_gauge!(
-                    "unsafe_max_priority_fee",
-                    global_fees.unsafe_max as u64,
-                    "account" => "none"
-                );
-                statsd_gauge!(
-                    "recommended_priority_fee",
-                    get_recommended_fee(global_fees) as u64,
-                    "account" => "none"
-                );
-                tokio::time::sleep(Duration::from_millis(1000)).await
+                let elapsed = start.elapsed().as_millis();
+                let remaining_time = 1_000u128.saturating_sub(elapsed);
+                match timeout(Duration::from_millis(remaining_time as u64), sampling_rxn.recv()).await
+                {
+                    Ok(res) => {
+                        match res {
+                            Some((accounts, include_vote, lookback_period)) =>
+                                priority_fee_tracker.record_specific_fees(accounts, include_vote, lookback_period),
+                            _ => {},
+                        }
+                    }
+                    Err(_) => {
+                        priority_fee_tracker.record_general_fees();
+                        start = Instant::now();
+                    },
+                };
             }
         });
     }
 
-    fn push_priority_fee_for_txn(
+    fn record_general_fees(&self) {
+        let global_fees = self.calculation1(&vec![], false, &None);
+        statsd_gauge!(
+            "min_priority_fee",
+            global_fees.min as u64,
+            "account" => "none"
+        );
+        statsd_gauge!("low_priority_fee", global_fees.low as u64, "account" => "none");
+        statsd_gauge!(
+            "medium_priority_fee",
+            global_fees.medium as u64,
+            "account" => "none"
+        );
+        statsd_gauge!(
+            "high_priority_fee",
+            global_fees.high as u64,
+            "account" => "none"
+        );
+        statsd_gauge!(
+            "very_high_priority_fee",
+            global_fees.very_high as u64,
+            "account" => "none"
+        );
+        statsd_gauge!(
+            "unsafe_max_priority_fee",
+            global_fees.unsafe_max as u64,
+            "account" => "none"
+        );
+        statsd_gauge!(
+            "recommended_priority_fee",
+            get_recommended_fee(global_fees) as u64,
+            "account" => "none"
+        );
+    }
+
+    fn record_specific_fees(&self, accounts: Vec<Pubkey>, include_vote: bool, lookback_period: Option<u32>)
+    {
+        let global_fees1 = self.calculation1(&accounts, include_vote, &lookback_period);
```

Comment:
can we call this something more descriptive, even old/new. And then I think it'll be easier to read if it was `new - old` since we expect new > old

Reply:
Yep, will rename to old and new